core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletMetadata.java

```diff
@@ -444,11 +444,18 @@ public TabletState getTabletState(Set<TServerInstance> liveTServers, TabletBalan
       return liveTServers.contains(future.getServerInstance()) ? TabletState.ASSIGNED
           : TabletState.ASSIGNED_TO_DEAD_SERVER;
     } else if (current != null) {
-
       if (liveTServers.contains(current.getServerInstance())) {
-        if (balancer != null && !balancer.isHostedInResourceGroup(conf,
-            new TabletServerIdImpl(current.getServerInstance()), currentTServerGrouping)) {
-          return TabletState.ASSIGNED_TO_WRONG_GROUP;
+        if (balancer != null) {
+          String resourceGroup = balancer.getResourceGroup(new TabletIdImpl(extent));
+          log.trace("Resource Group for extent {} is {}", extent, resourceGroup);
+          Set<TabletServerId> tservers = currentTServerGrouping.get(resourceGroup);
+          if (tservers == null) {
+            log.warn("No TabletServers for resource group {} ", resourceGroup);
```

Comment:
his warn logging could get really spammy. Also for this case it seems like it should return TabletState.ASSIGNED_TO_WRONG_GROUP. The tserver would be unassigned and then left unassigned until some tservers showed up in the group.

Reply:
Resolved in 899cb4e